src/main/java/kg/jarkyn/server/Server.java

```diff
@@ -0,0 +1,33 @@
+package kg.jarkyn.server;
+
+import kg.jarkyn.server.Incoming.Requester;
+import kg.jarkyn.server.Outgoing.Responder;
+import kg.jarkyn.server.Outgoing.Response;
+
+public class Server {
+    private final ReceivingSocket serverSocket;
+    private final Responder responder;
+
+    public Server(ReceivingSocket serverSocket, Responder responder) {
```

Comment:
I think the naming issue is visible here ;)
The class name is `ReceivingSocket` yet the argument name is `serverSocket`. Maybe `ReceivingSocket` actually wants to be `ServerSocket`.